src/parser/types.rs

```diff
@@ -419,7 +437,12 @@ impl Ty {
                                     ctx.error(ErrorKind::InvalidArraySize, siz_first.span)
                                 })?)
                             }
-                            TokenKind::Generic(name) => {
```

Comment:
personally I'm not a fan of this branch :D I would be happier if we didn't allow operating on generics in types (like in Rust) and maybe we can try to have this in a later PR. But this monomorphization change is trying to do too many things already IMO